Calmspace/PracticeTableViewController.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  PracticeTableViewController.swift
+//  Calmspace
+//
+//  Created by Admin on 09/10/2023.
+//
+
+import UIKit
+import Foundation
+    
+    var practiceSections = ["Morning Meditation", "Daily Meditation", "Before Sleep Meditation", "Yoga: Beginner & All Levels", "Yoga: Medium & Advance Levels"]
+    var practiceData: Practice!
+    
+    struct Practice: Codable {
+        let dailyMeditation: [Meditation]
+        let yogaBeginnerAllLevels: [YogaSession]
+        let yogaMediumAdvanceLevels: [YogaSession]
+        let morningMeditation: [Meditation]
+        let beforeSleepMeditation: [Meditation]
+        
+    enum CodingKeys: String, CodingKey {
+        case dailyMeditation = "Daily Meditation"
+        case yogaBeginnerAllLevels = "Yoga: Beginner & All Levels"
+        case yogaMediumAdvanceLevels = "Yoga: Medium & Advance Levels"
+        case morningMeditation = "Morning Meditation"
+        case beforeSleepMeditation = "Before Sleep Meditation"
+        }
+    }
+    struct Meditation: Codable {
+        let title, description: String
+        let link: URL
+        
+    enum CodingKeys: String, CodingKey {
+        case title = "Title"
+        case link = "URL"
+        case description = "Description"
+        }
+    }
+    
+    struct YogaSession: Codable {
+        let title, description: String
+        let link: URL
+        
+    enum CodingKeys: String, CodingKey {
+        case title = "Title"
+        case link = "URL"
+        case description = "Description"
+        }
+    }
+
+class PracticeTableViewController: UITableViewController {
+    
+    var meditationSections = ["Morning Meditation", "Daily Meditation", "Before Sleep Meditation"]
+    var yogaSections = ["Yoga: Beginner & All Levels", "Yoga: Medium & Advance Levels"]
+    var showMeditationSections = true
+    
+    override func viewDidLoad() {
+        
+        super.viewDidLoad()
+        print("PracticeTableViewController - View Did Load")
+        
+        guard let url = Bundle.main.url(forResource: "Practice", withExtension: "json"),
+              let data = try? Data(contentsOf: url) else {
+            fatalError("Failed to load Practice.json")
+        }
+        
+        let decoder = JSONDecoder()
+        do {
+            practiceData = try decoder.decode(Practice.self, from: data)
+        } catch {
+            fatalError("Failed to decode Practice.json: \(error)")
+        }
+        
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if showMeditationSections {
```

Comment:
Having a boolean value to know what data to return might be not scalable. What happens when there's the third table? What we can do to improve it:
1. Have a separate data source for each table
2. Distinguish between tables using the first parameter of this function, `tableView`. You can create outlets to each table view, and then compare their addresses to the address of the parameter, e.g.:
```swift
@IBoutlet private var meditationTable: UITableView
@IBoutlet private var yogaTable: UITableView

...

override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    if tableView === meditationTable {
        return meditationSections.count
    }
    return yogaSections.count
}
```

We are taking advantage of two things:

1. The `===` operator allows to compare the addresses of the objects. 
2. The first parameter in delegate calls is always the object that is calling the given method.

Does it make sense?